tests/integration/standard/test_policies.py

```diff
@@ -0,0 +1,94 @@
+
+    def make_query_plan(self, working_keyspace=None, query=None):
+        pos = self._position
+        self._position += 1
+
+        hosts = []
+        for _ in range(10):
+            hosts.extend(self._live_hosts)
+
+        return hosts
+
+
+class SpecExectTest(BasicSharedKeyspaceUnitTestCase):
+
+    def setUp(self):
+        spec_ep_brr = ExecutionProfile(load_balancing_policy=BadRoundRobinPolicy(), speculative_execution_policy=ConstantSpeculativeExecutionPolicy(.01, 20))
+        spec_ep_rr = ExecutionProfile(speculative_execution_policy=ConstantSpeculativeExecutionPolicy(.01, 20))
+        spec_ep_rr_lim = ExecutionProfile(load_balancing_policy=BadRoundRobinPolicy(), speculative_execution_policy=ConstantSpeculativeExecutionPolicy(.01, 1))
+        self.cluster.add_execution_profile("spec_ep_brr", spec_ep_brr)
+        self.cluster.add_execution_profile("spec_ep_rr", spec_ep_rr)
+        self.cluster.add_execution_profile("spec_ep_rr_lim", spec_ep_rr_lim)
+
+    def test_speculative_execution(self):
+        """
+        Test to ensure that speculative execution honors LBP, and that they retry appropriately.
+
+        This test will use various LBP, and ConstantSpeculativeExecutionPolicy settings and ensure the proper number of hosts are queried
+        @since 3.7.0
+        @jira_ticket PYTHON-218
+        @expected_result speculative retries should honor max retries, idempotent state of queries, and underlying lbp.
+
+        @test_category metadata
+        """
+        self.session.execute("""USE {0}""".format(self.keyspace_name))
+        self.session.execute("""create or replace function timeout (arg int) RETURNS NULL ON NULL INPUT RETURNS int LANGUAGE java AS $$ long start = System.currentTimeMillis(); while(System.currentTimeMillis() - start < arg){} return arg; $$;""")
+        self.session.execute("""CREATE TABLE  d (k int PRIMARY KEY , i int);""")
+        self.session.execute("""INSERT INTO d (k,i) VALUES (0, 1000);""")
+        statement = SimpleStatement("""SELECT timeout(i) FROM d WHERE k =0""", is_idempotent=True)
+        statement_non_idem = SimpleStatement("""SELECT timeout(i) FROM d WHERE k =0""", is_idempotent=False)
+
+        # This LBP should repeat hosts up to around 30
+        result = self.session.execute(statement, execution_profile='spec_ep_brr')
+        self.assertEqual(21, len(result.response_future.attempted_hosts))
+
+        # This LBP should keep host list to 3
+        result = self.session.execute(statement, execution_profile='spec_ep_rr')
+        self.assertEqual(3, len(result.response_future.attempted_hosts))
+        # Spec_execution policy should limit retries to 1
+        result = self.session.execute(statement, execution_profile='spec_ep_rr_lim')
+
+        self.assertEqual(2, len(result.response_future.attempted_hosts))
+        # Spec_execution policy should not be used if  the query is not idempotent
+        result = self.session.execute(statement_non_idem)
```

Comment:
I think this is using default profile, which does not have spec exec. Do we want to add a test for non-idempotent+spec exec policy?